### PR TITLE
sched: Remove KEEP_ALIVE_HACK hack

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
See the discussion here: https://github.com/apache/incubator-nuttx/pull/6660

## Impact
Save the power in the round robin and sporadic scheduler

### CODE REVIEW

File: sched/sched/sched_timerexpiration.c [hunk 1]
```diff
@@ -233,15 +212,6 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
 
   /* Returning zero means that there is no interesting event to be timed */
 
-#ifdef KEEP_ALIVE_HACK
```
Comment (pkarashchenko) [inline sched/sched/sched_timerexpiration.c:236]: Just for reference: https://github.com/apache/incubator-nuttx/commit/6546fa39c7a#diff-ec549151b9b686bd3d14a18d3a6a0ca940987bff9b59ab299166ea506f604219

File: sched/sched/sched_timerexpiration.c [hunk 2]
```diff
@@ -233,15 +212,6 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
...
-  if (ret == 0)
-    {
-      /* Apply the keep alive hack */
-
-      return KEEP_ALIVE_TICKS;
```
Comment (mkeydevelop) [inline sched/sched/sched_timerexpiration.c:241]: Time slice not work on sporadic?

**Review by hartmannathan — COMMENTED** [review]
So, does this depend on [PR-6660](https://github.com/apache/incubator-nuttx/pull/6660)?

**Comment by acassis** [comment]
@xiaoxiang781216 is the some testing to guarantee that sporadic scheduling is working? I found a good explanation here: https://www.embedded.com/scheduling-sporadic-events/

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 is the some testing to guarantee that sporadic scheduling is working? I found a good explanation here: https://www.embedded.com/scheduling-sporadic-events/

No, we never use the sporadic scheduler.

**Pull request closed without merging**